utilities/general-helper-functions.metta

```diff
@@ -46,4 +46,90 @@
         )
 )
 )
+)
+
+(= (getchildren $expr $literals $nonLiterals)
+    (if (== $expr ())
+        ($literals $nonLiterals)
+        (let* 
+            (   
+                ($head (car-atom $expr))
+                ($tail (cdr-atom $expr))
+                ($isLiteral (or (== (get-metatype $head) Symbol) (unify $head (NOT $_) True False)))
+            )
+            (if (or (== $head AND) (== $head OR)) 
+                (getchildren $tail $literals $nonLiterals)
+
+                (if $isLiteral 
+                    (getchildren $tail (concatTuple $literals ($head)) $nonLiterals)
+                    (getchildren $tail $literals (concatTuple $nonLiterals ($head)))        
+                )
+            )
+        )
+    )
+)
+
+(= (getGsetAndChildren $expr) (getchildren $expr () ()))
+
+(= (findIntersection $tuple1 $tuple2)
+    (if (== $tuple1 ())
+        () 
+        (let* 
+            (
+                ($head (car-atom $tuple1))
+                ($tail (cdr-atom $tuple1))
+            )
+            (if (isMember $tuple2 $head)
+                (let $val (findIntersection $tail $tuple2) (cons-atom $head $val))
+                (findIntersection $tail $tuple2)
+            )
+        )
+    )
+)
+
+(= (findCommonLiterals $literals $children)
```

Comment:
Best if you move this file back to the `promote-common-constraints` file, since it's highly unlikely to get reused. If it does we will take it out :)